### DISCUSSION TITLE
Versioning

### DISCUSSION THREAD

**Discussion by bilogic** [laravel-zero/laravel-zero#370] [Q&A] [answered]
Hi,

What is `app("git.version")` mean to do? Where does it get the version from? 

I tried the following with no effect, `app:build`'s default value still showed as `unreleased`

1. git's tag
2. `composer.json`'s `version`

Thank you.

**Comment by owenvoke** [comment, marked as answer]
This is registered via the [`GitVersionServiceProvider`](https://github.com/laravel-zero/framework/blob/master/src/Providers/GitVersion/GitVersionServiceProvider.php#L25), which uses `git describe --tags --abbrev=0` to try and get the latest tag. As far as I know, this is for non-compiled applications, such as if you clone the repository. For `app:build`, I've always tended to provide a `--build-version=` flag on the command usage and manually specified this.

**Reply by bilogic** [reply]
> As far as I know, this is for non-compiled applications...

My expectation was for `app:build` to retrieve git's tag, perhaps can make this change.

**Reply by owenvoke** [reply]
I'm not sure if this would be possible, as the compiled application (at least from everywhere I've seen it used) is usually committed to the repository before the tag is created. I guess if you weren't doing that, and for example, were creating a tag and then publishing a Phar alongside with GitHub actions, you'd just read the tag and pass it in with `--build-version=${{ github.ref_name }}` or something similar

**Reply by bilogic** [reply]
Oh, my process would be to commit + tag, then compile. 

I will try out the `${{ github.ref_name }}`, but where would this come from?
Some environment variable? Sorry, I'm not that savvy with Linux.
